packages/flet/lib/src/controls/page.dart

```diff
@@ -1131,14 +1132,28 @@ class _ViewControlState extends State<ViewControl> with FletStoreMixin {
                 child: scaffold,
               );
             }
-
-            return Directionality(
+            var result = Directionality(
                 textDirection: textDirection,
                 child: widget.loadingPage != null
                     ? Stack(
                         children: [scaffold, widget.loadingPage!],
                       )
                     : scaffold);
+            return withPageArgs((context, pageArgs) {
+              var backgroundDecoration = parseBoxDecoration(
+                  Theme.of(context), control, "decoration", pageArgs);
+              var foregroundDecoration = parseBoxDecoration(
+                  Theme.of(context), control, "foregroundDecoration", pageArgs);
+              if (backgroundDecoration != null ||
+                  foregroundDecoration != null) {
+                return Container(
+                  decoration: backgroundDecoration,
+                  foregroundDecoration: foregroundDecoration,
+                  child: result,
+                );
+              }
+              return result;
+            });
```

Comment:
I'd make wrapping into Container conditional, i.e. when the decoration is specified on Python side. Just to be on the safe side.

Reply:
Legit 👍🏾

Reply:
Ah, I see, nvm. I was probably expecting `else` for clarity.